includes/chart.js

```diff
@@ -643,13 +663,41 @@ function deleteConnection(event) {
     // If no such line, try finding a line the other way around
     if (!line) {
         line = document.getElementById(`connector_${destinationRowIdNum}-${startRowIdNum}`);
+        if (!line) {
+            return;  // Not valid in reverse either? Just exit.
+        }
+
+        // reverse line was found, so swap INA.startShapeId and destinationShapeId
+        [INA.startShapeId, destinationShapeId] = [destinationShapeId, INA.startShapeId]
     }
 
-    if (line) {
-        line.remove();
+    // Remove line from session tracking
+    let connectionColor = line.getAttribute("stroke");
+    let connectionIdx = INA.connections.findIndex(
+        (connection) => doArraysMatch(
+            connection,
+            [INA.startShapeId, destinationShapeId, connectionColor]
+        )
+    );
+    INA.connections.splice(connectionIdx, 1)  // splice(idx, n): remove n elements starting at idx
+    storeDatainSession();
```

Comment:
IIUC, shouldn't you check the return value, or maybe `sessionState`?  Or maybe `success` and `error` are not quite correct here:

https://github.com/ESI-FAR/INA-tool/blob/2a46b5689ef833a19440d389b3e2bc57e83eeaca/includes/uploader.js#L230-L248

If this is a bit complicated, then maybe create an issue an address this later?

Reply:
Apart from printing the result in the console log, we don't do anything else with these errors yet, so I think that's a larger scope question to tackle. I'll make a new issue for it instead.